interp/smartlocate.mli

```diff
@@ -39,13 +40,13 @@ val global_inductive_with_alias : qualid -> inductive
 val global_constructor_with_alias : qualid -> constructor
 
 (** Locate a reference taking into account notations and "aliases" *)
-val smart_global : ?head:bool -> qualid Constrexpr.or_by_notation -> GlobRef.t
+val smart_global : ?head:bool -> (glob_constr -> Pp.t) -> qualid Constrexpr.or_by_notation -> GlobRef.t
```

Comment:
Not a big fan of that change. It seems to me that this makes that API harder to use for a relatively accessory feature. I understand some ml file dependency imposes that design, but isn't there any alternative? At worst, instead of printing the multiple notations, the error message could just say something like "use Locate "ntn" to list the <n> possible interpretations." At the very least, we should add a comment in `smartlocate.mli` explaining what the printer is and how to get it.

Reply:
Why not raise an algebraic exception and register the printer later on? That seems to be the best practice.

Reply:
OK, will do. Do I put the handler in `himsg.ml`?

Reply:
This looks like a reasonable place indeed.

Reply:
Done, much less intrusive!